## <font color="navy blue"> 🌍🎓  𝐈𝐧𝐭𝐞𝐫𝐧𝐚𝐭𝐢𝐨𝐧𝐚𝐥 𝐄𝐝𝐮𝐜𝐚𝐭𝐢𝐨𝐧 𝐂𝐨𝐬𝐭: 𝐀 𝐂𝐨𝐦𝐩𝐫𝐞𝐡𝐞𝐧𝐬𝐢𝐯𝐞 𝐒𝐐𝐋 𝐒𝐭𝐮𝐝𝐲

###  <font color="191970"> 💻🔧 𝐏𝐚𝐜𝐤𝐚𝐠𝐞 𝐈𝐧𝐬𝐭𝐚𝐥𝐥𝐚𝐭𝐢𝐨𝐧

In [5]:
! pip install sqlalchemy 
! pip install pandasql


[notice] A new release of pip is available: 23.0.1 -> 25.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip



[notice] A new release of pip is available: 23.0.1 -> 25.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip


###  <font color="191970"> 📚 𝐈𝐦𝐩𝐨𝐫𝐭𝐢𝐧𝐠 𝐋𝐢𝐛𝐫𝐚𝐫𝐢𝐞𝐬

In [13]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns

from pandasql import sqldf 

In [14]:
df=pd.read_csv("International_Education_Costs.csv")
df

Country        City                      University  \
0             USA   Cambridge              Harvard University   
1              UK      London         Imperial College London   
2          Canada     Toronto           University of Toronto   
3       Australia   Melbourne         University of Melbourne   
4         Germany      Munich  Technical University of Munich   
..            ...         ...                             ...   
902        France  Strasbourg        University of Strasbourg   
903      Malaysia       Nilai                            USIM   
904  Saudi Arabia     Al-Ahsa          King Faisal University   
905           USA     Seattle        University of Washington   
906            UK  Nottingham        University of Nottingham   

                    Program     Level  Duration_Years  Tuition_USD  \
0          Computer Science    Master             2.0        55400   
1              Data Science    Master             1.0        41200   
2        Business Analytics    Master             2.0        38500   
3               Engineering    Master             2.0        42000   
4    Mechanical Engineering    Master             2.0          500   
..                      ...       ...             ...          ...   
902          Data Analytics    Master             2.0         4000   
903        Computer Science  Bachelor             3.0         6800   
904     Information Systems    Master             2.0         4200   
905    Software Development       PhD             5.0        50000   
906        Data Engineering    Master             2.0        34000   

     Living_Cost_Index  Rent_USD  Visa_Fee_USD  Insurance_USD  Exchange_Rate  
0                 83.5      2200           160           1500           1.00  
1                 75.8      1800           485            800           0.79  
2                 72.5      1600           235            900           1.35  
3                 71.2      1400           450            650           1.52  
4                 70.5      1100            75            550           0.92  
..                 ...       ...           ...            ...            ...  
902               70.2      1000            99            850           0.92  
903               50.5       400           120            400           4.65  
904               64.2       600           200            800           3.75  
905               77.8      2000           160           1500           1.00  
906               61.2       800           485            800           0.79  

[907 rows x 12 columns]

## <font color="navy blue"> 1. 💵 Total Cost Estimation
### <font color="191970"> Calculate total estimated cost of education for students

In [15]:
pysqldf = lambda q: sqldf(q, globals())

total_cost = """
SELECT DISTINCT Country, University,Level, Program, Duration_Years, Tuition_USD,
       Total_Rent_Cost, Visa_Fee_USD, Insurance_USD, Total_Cost_USD
FROM (
    SELECT
      Country,
      Level,
      University,
      Program,
      Duration_Years,
      Tuition_USD,
      (Rent_USD * Duration_Years * 12) AS Total_Rent_Cost,
      Visa_Fee_USD,
      Insurance_USD,
      (Tuition_USD + (Rent_USD * Duration_Years * 12) + Visa_Fee_USD + Insurance_USD) AS Total_Cost_USD
    FROM
      df
)

ORDER BY
  Country asc;

  """
result = pysqldf(total_cost)
result

Country                            University     Level  \
0       Algeria  University of Science and Technology  Bachelor   
1       Algeria                    University of Oran    Master   
2       Algeria             University of Constantine       PhD   
3       Algeria              Badji Mokhtar University    Master   
4       Algeria                 University of Tlemcen  Bachelor   
..          ...                                   ...       ...   
902  Uzbekistan            Samarkand State University    Master   
903  Uzbekistan              Bukhara State University  Bachelor   
904  Uzbekistan                    Namangan Institute    Master   
905  Uzbekistan              Andijan State University    Master   
906     Vietnam                                 HCMUT  Bachelor   

                    Program  Duration_Years  Tuition_USD  Total_Rent_Cost  \
0          Computer Science             3.0         1200           9000.0   
1      Computer Engineering             2.0         1100           5280.0   
2            Data Analytics             4.0         1300           9600.0   
3      Computer Engineering             2.0         1000           4320.0   
4       Information Systems             3.0          900           5760.0   
..                      ...             ...          ...              ...   
902          Data Analytics             2.0         3800           6000.0   
903        Computer Science             4.0         3500          10560.0   
904  Information Technology             2.0         3400           4800.0   
905    Software Development             2.0         3200           4320.0   
906    Software Engineering             4.0         3200          21600.0   

     Visa_Fee_USD  Insurance_USD  Total_Cost_USD  
0              80            200         10480.0  
1              80            200          6660.0  
2              80            200         11180.0  
3              80            200          5600.0  
4              80            200          6940.0  
..            ...            ...             ...  
902           160            250         10210.0  
903           160            250         14470.0  
904           160            250          8610.0  
905           160            250          7930.0  
906           120            250         25170.0  

[907 rows x 10 columns]

🔍 Findings:
1. Countries show wide variation in total study cost. Countries like Germany or France tend to have lower tuition,    while USA or UK have higher total costs due to both tuition and living expenses.
2. Longer programs (e.g., Bachelor's) significantly increase rent and insurance costs, even if tuition is low.
3. Postgraduate programs often have shorter durations but higher tuition. Undergraduate programs have higher          living expenses due to extended duration.

##  <font color="navy blue">💲 Cheapest Countries for a Program
### <font color="191970"> Find the most affordable country for "Data Analytics" program (based on total cost)

In [16]:
pysqldf = lambda q: sqldf(q, globals())

cheapest_program = """
SELECT
  Country,
  University,
  Level,
  Program,
  Duration_Years,
  Tuition_USD,
  (Rent_USD * Duration_Years * 12) AS Rent_Cost,
  (Tuition_USD + (Rent_USD * Duration_Years * 12) + Visa_Fee_USD + Insurance_USD) AS Total_Cost
FROM
  df
WHERE
  Program = 'Data Analytics'
ORDER BY
  Total_Cost ASC
LIMIT 5;

   """
result = pysqldf(cheapest_program)
result

Country                     University   Level         Program  \
0       Egypt               Aswan University  Master  Data Analytics   
1     Morocco  University of Al Quaraouiyine  Master  Data Analytics   
2     Romania             Ovidius University  Master  Data Analytics   
3  Uzbekistan     Samarkand State University  Master  Data Analytics   
4     Ukraine             Dnipro Polytechnic  Master  Data Analytics   

   Duration_Years  Tuition_USD  Rent_Cost  Total_Cost  
0             2.0         1900     3840.0      6040.0  
1             2.0         1800     6000.0      8120.0  
2             2.0         1900     7680.0     10050.0  
3             2.0         3800     6000.0     10210.0  
4             2.0         3300     7200.0     10920.0

🔍 Findings:
1. The query successfully highlights the 5 least expensive Data Analytics programs, providing a clear list for        budget-conscious students.
2. The result helps identify countries/universities that offer value-for-money education in a high-demand field       like Data Analytics.
3. These universities are typically located in countries with:

   Lower tuition fees   ,    Affordable rent   and    Lower visa and insurance charges



### <font color="navy blue"> 3. 🔢 Creating table "universities" , "programs" and "costs"

In [17]:
universities = df[['University', 'City', 'Country']].drop_duplicates().reset_index(drop=True)

In [18]:
universities.insert(0, 'University_ID', universities.index + 1)

In [19]:
pysqldf = lambda q: sqldf(q, globals())

# Preview the Universities table
print(universities)

     University_ID                      University           City  \
0                1              Harvard University      Cambridge   
1                2         Imperial College London         London   
2                3           University of Toronto        Toronto   
3                4         University of Melbourne      Melbourne   
4                5  Technical University of Munich         Munich   
..             ...                             ...            ...   
631            632                            UiTM      Shah Alam   
632            633          Umm Al-Qura University         Makkah   
633            634             Stanford University  San Francisco   
634            635                            USIM          Nilai   
635            636          King Faisal University        Al-Ahsa   

          Country  
0             USA  
1              UK  
2          Canada  
3       Australia  
4         Germany  
..            ...  
631      Malaysia  
632  Saudi 

In [20]:
programs = df[['Program', 'Level', 'Duration_Years']].drop_duplicates().reset_index(drop=True)
programs.insert(0, 'Program_ID', programs.index + 1)

In [21]:
df_costs = df.merge(universities, on=['University', 'City', 'Country'], how='left')
df_costs = df_costs.merge(programs, on=['Program', 'Level', 'Duration_Years'], how='left')

In [22]:
costs = df_costs[['University_ID', 'Program_ID', 'Tuition_USD', 'Rent_USD', 'Visa_Fee_USD', 'Insurance_USD','Living_Cost_Index','Exchange_Rate']]


🔍 Findings:
1. Finland, Greece, Brazil, and Argentina offer tuition-free Data Science programs at reputable public universities.
These are excellent choices for students seeking high-quality education with minimal financial burden.

2. Universities like Columbia ($62,000), University of Chicago  ($58,000), and Carnegie Mellon ($56,000) represent the upper end of the cost spectrum.
While prestigious, these programs are significantly more expensive.

## <font color="navy blue"> 4. 🏳️ Programs Offered in a Specific Country
### <font color="191970"> Find all programs in USA

In [23]:
pysqldf = lambda q: sqldf(q, globals())

programs_offered = """
SELECT
  Country,
  u.University,
  p.Program,
  p.Level
FROM
  programs p
JOIN costs co ON co.program_ID = p.program_ID
JOIN universities u ON u.University_ID = co.University_ID
WHERE
  u.Country = 'USA';
  """
result = pysqldf(programs_offered)
result

Country                 University                  Program     Level
0      USA         Harvard University         Computer Science    Master
1      USA     University of Michigan    Aerospace Engineering  Bachelor
2      USA     University of Michigan         Computer Science  Bachelor
3      USA               Georgia Tech   Industrial Engineering  Bachelor
4      USA               Georgia Tech         Computer Science  Bachelor
..     ...                        ...                      ...       ...
73     USA       University of Denver           Data Analytics    Master
74     USA  Portland State University             Data Science    Master
75     USA         Harvard University  Artificial Intelligence    Master
76     USA              Massachusetts                 Robotics       PhD
77     USA        Stanford University             Data Science    Master

[78 rows x 4 columns]

🔍 Findings:

1. The U.S. provides extensive academic opportunities in a variety of high-demand fields, particularly in STEM and    data-related disciplines, across all degree levels. This makes it a highly versatile and attractive destination
   for both undergraduate and graduate international students.

3. Programs span across Bachelor’s, Master’s, and PhD levels, indicating the U.S. offers complete academic          pathways in many fields.

## <font color="navy blue"> 5. ⏱️ Duration of Programs according to countries and level

In [48]:
pysqldf = lambda q: sqldf(q, globals())

duration = """

SELECT
University, Country , Program , Level, Duration_Years
FROM
  programs p
JOIN costs co ON p.Program_ID = co.Program_ID
JOIN universities u ON u.University_ID = co.University_ID
WHERE
  p.Program = 'Data Analytics' 
  group by Country,level;
  """

time = pysqldf(duration)
time

University      Country         Program     Level  \
0         University of Constantine      Algeria  Data Analytics       PhD   
1       National University of Cuyo    Argentina  Data Analytics       PhD   
2             James Cook University    Australia  Data Analytics    Master   
3            University of Salzburg      Austria  Data Analytics  Bachelor   
4        University of Saskatchewan       Canada  Data Analytics    Master   
5                 Aarhus University      Denmark  Data Analytics    Master   
6                  Aswan University        Egypt  Data Analytics    Master   
7               Sorbonne University       France  Data Analytics    Master   
8               University of Ghana        Ghana  Data Analytics    Master   
9               University of Crete       Greece  Data Analytics    Master   
10      John von Neumann University      Hungary  Data Analytics       PhD   
11           University of Akureyri      Iceland  Data Analytics  Bachelor   
12                    IIT Hyderabad        India  Data Analytics       PhD   
13  Sharif University of Technology         Iran  Data Analytics    Master   
14           Trinity College Dublin      Ireland  Data Analytics    Master   
15            Politecnico di Milano        Italy  Data Analytics    Master   
16          Sacred Heart University   Luxembourg  Data Analytics    Master   
17                             UiTM     Malaysia  Data Analytics    Master   
18         Tecnologico de Monterrey       Mexico  Data Analytics  Bachelor   
19    University of Al Quaraouiyine      Morocco  Data Analytics    Master   
20              Victoria University  New Zealand  Data Analytics    Master   
21    Quality Leadership University       Panama  Data Analytics  Bachelor   
22                             PUCP         Peru  Data Analytics    Master   
23               Ovidius University      Romania  Data Analytics    Master   
24          Ural Federal University       Russia  Data Analytics       PhD   
25  Singapore Management University    Singapore  Data Analytics  Bachelor   
26    Kyungpook National University  South Korea  Data Analytics    Master   
27         University of St. Gallen  Switzerland  Data Analytics    Master   
28              University of Dubai          UAE  Data Analytics    Master   
29               Cardiff University           UK  Data Analytics    Master   
30            University of Chicago          USA  Data Analytics    Master   
31               Dnipro Polytechnic      Ukraine  Data Analytics    Master   
32       Samarkand State University   Uzbekistan  Data Analytics    Master   

    Duration_Years  
0              4.0  
1              4.0  
2              2.0  
3              3.0  
4              2.0  
5              2.0  
6              2.0  
7              2.0  
8              2.0  
9              2.0  
10             4.0  
11             3.0  
12             4.0  
13             2.0  
14             1.0  
15             2.0  
16             2.0  
17             2.0  
18             4.0  
19             2.0  
20             2.0  
21             3.0  
22             2.0  
23             2.0  
24             4.0  
25             4.0  
26             2.0  
27             2.0  
28             2.0  
29             1.0  
30             2.0  
31             2.0  
32             2.0

## <font color="navy blue"> 6. 🏛️Number of Universities in each Country

In [158]:
pysqldf = lambda q: sqldf(q, globals())

query = """

SELECT
  Country,
  COUNT(*) AS Num_Universities
FROM
  universities
GROUP BY
  Country
ORDER BY
  Num_Universities DESC;
"""
pysqldf(query)


Country  Num_Universities
0               UK                57
1              USA                43
2           Canada                27
3        Australia                23
4          Germany                22
..             ...               ...
66    South Africa                 1
67         Nigeria                 1
68          Israel                 1
69       Hong Kong                 1
70  Czech Republic                 1

[71 rows x 2 columns]

🔍 Findings:

The dataset shows a concentration of universities in Western, developed countries, especially the UK and USA, which dominate the list. This reflects their global prominence in higher education, though it also suggests a potential underrepresentation of universities from developing regions.



 ## <font color="navy blue"> 7. 💰 Average tutition fee by Country and level

In [28]:
pysqldf = lambda q: sqldf(q, globals())

Avg_fee = """

SELECT
University, Country , Program , Level, avg(Tuition_USD) as avg_tuition_fee
FROM
  programs p
JOIN costs co ON p.Program_ID = co.Program_ID
JOIN universities u ON u.University_ID = co.University_ID
group by Country,level;
  """
pysqldf(Avg_fee)


University     Country                 Program  \
0    University of Science and Technology     Algeria        Computer Science   
1                      University of Oran     Algeria    Computer Engineering   
2               University of Constantine     Algeria          Data Analytics   
3              University of Buenos Aires   Argentina    Software Engineering   
4          National University of Cordoba   Argentina            Data Science   
..                                    ...         ...                     ...   
182               Universidad ORT Uruguay     Uruguay        Computer Science   
183           Universidad de la Republica     Uruguay    Computer Engineering   
184              INHA University Tashkent  Uzbekistan        Computer Science   
185                    Namangan Institute  Uzbekistan  Information Technology   
186                                 HCMUT     Vietnam    Software Engineering   

        Level  avg_tuition_fee  
0    Bachelor      1050.000000  
1      Master      1050.000000  
2         PhD      1300.000000  
3    Bachelor         0.000000  
4      Master         0.000000  
..        ...              ...  
182    Master      4500.000000  
183       PhD         0.000000  
184  Bachelor      3850.000000  
185    Master      3466.666667  
186  Bachelor      3200.000000  

[187 rows x 5 columns]

## <font color="navy blue"> 8. 💸Total Estimated Monthly Expense
### <font color="191970">  Find monthly expenditure

In [22]:
pysqldf = lambda q: sqldf(q, globals())
monthly_expense = """

SELECT
  u.University,
  u.Country,
  u.City,
  p.Program,
  ROUND((co.Rent_USD + (co.Insurance_USD / (p.Duration_Years * 12))), 2) AS Estimated_Monthly_Expense
FROM
  costs co
JOIN universities u ON co.University_ID = u.University_ID
JOIN programs p ON co.Program_ID = p.Program_ID
ORDER BY
  Estimated_Monthly_Expense ASC
LIMIT 10;
"""
pysqldf(monthly_expense)

University     Country       City                  Program  \
0  University of Carthage     Tunisia    Bizerte  Artificial Intelligence   
1                    SUST  Bangladesh     Sylhet     Computer Engineering   
2   University of Tlemcen     Algeria    Tlemcen      Information Systems   
3        Aswan University       Egypt      Aswan           Data Analytics   
4     University of Gabes     Tunisia      Gabes     Computer Engineering   
5     Mansoura University       Egypt   Mansoura      Information Systems   
6  University of Monastir     Tunisia   Monastir             Data Science   
7                    RUET  Bangladesh   Rajshahi     Software Engineering   
8    University of Sousse     Tunisia     Sousse         Computer Systems   
9    Port Said University       Egypt  Port Said         Computer Science   

   Estimated_Monthly_Expense  
0                     155.56  
1                     164.17  
2                     165.56  
3                     168.33  
4                     168.33  
5                     174.17  
6                     174.17  
7                     178.33  
8                     185.56  
9                     188.33

🔍 Findings:

1. Students seeking budget-friendly international education can strongly consider universities in Tunisia, Egypt,     Bangladesh, and Algeria, where monthly living expenses remain under $200 USD—an attractive option for both         local and global students.

2. These affordable universities offer diverse programs, including Artificial Intelligence, Computer Science,         Information Systems, and Data Analytics, showing that cost-effectiveness does not compromise program choice.

3. Countries like Tunisia, Egypt, and Bangladesh consistently appear in the top 10, reflecting affordable housing     and insurance rates.

## <font color="navy blue">  9. 📈 Top 5 Most Expensive Programs by Total Cost
### <font color="191970"> Find Countries where the cost of living and studying is Expensive

In [19]:
pysqldf = lambda q: sqldf(q, globals())
expensive = """

SELECT
  u.University,
  u.Country,
  p.Program,
  Tuition_USD,
  (co.Tuition_USD + co.Rent_USD * p.Duration_Years * 12 + co.Visa_Fee_USD + co.Insurance_USD) AS Total_Cost
FROM
  programs p
JOIN costs co ON p.Program_ID = co.Program_ID
JOIN universities u ON u.University_ID = co.University_ID
ORDER BY
  Total_Cost DESC
LIMIT 5;
"""
pysqldf(expensive)

University Country                 Program  Tuition_USD  \
0  Stanford University     USA        Computer Science        54200   
1  Stanford University     USA        Computer Science        52000   
2  Stanford University     USA                 Physics        52000   
3                  MIT     USA  Electrical Engineering        53800   
4                 UCLA     USA    Computer Engineering        52000   

   Total_Cost  
0    193860.0  
1    191660.0  
2    191660.0  
3    187460.0  
4    185660.0

🔍 Findings:

1. Stanford University (USA) dominates the list of most expensive programs, with Computer Science and Physics         programs incurring total costs over $190,000 USD.

2. Top-tier U.S. universities not only charge premium tuition fees but also have significant additional living and    administrative costs, making them some of the costliest education destinations globally.

3. These high costs are due to a combination of high tuition fees, long program durations, and substantial            living/visa/insurance expenses.

## <font color="navy blue"> 10.⏳Programs with the Longest Duration
### <font color="191970"> Identify programs that require the most time to complete.

In [22]:
pysqldf = lambda q: sqldf(q, globals())
longest_duration = """

SELECT
  p.Program,
  p.Level,
  p.Duration_Years,
  u.University,
  u.Country
FROM
  programs p
JOIN costs co ON co.Program_ID = p.Program_ID
JOIN universities u ON co.University_ID = u.University_ID
WHERE Duration_Years = 5.0 ;
"""
pysqldf(longest_duration)

Program     Level  Duration_Years  \
0          Computer Science       PhD             5.0   
1          Computer Science       PhD             5.0   
2          Computer Science       PhD             5.0   
3          Computer Science       PhD             5.0   
4          Computer Science       PhD             5.0   
5          Computer Science       PhD             5.0   
6    Electrical Engineering       PhD             5.0   
7   Artificial Intelligence       PhD             5.0   
8      Software Engineering  Bachelor             5.0   
9                   Physics       PhD             5.0   
10              Engineering       PhD             5.0   
11                Chemistry       PhD             5.0   
12                 Robotics       PhD             5.0   
13                 Robotics       PhD             5.0   
14         Machine Learning       PhD             5.0   
15     Computer Engineering       PhD             5.0   
16     Software Development       PhD             5.0   

                    University    Country  
0          Stanford University        USA  
1          Stanford University        USA  
2     University of Washington        USA  
3      University of Wisconsin        USA  
4                         UIUC        USA  
5           Cornell University        USA  
6                          MIT        USA  
7                Johns Hopkins        USA  
8   University of Buenos Aires  Argentina  
9          Stanford University        USA  
10                Georgia Tech        USA  
11                UC San Diego        USA  
12                   UT Austin        USA  
13               Massachusetts        USA  
14                Georgia Tech        USA  
15                        UCLA        USA  
16    University of Washington        USA

## <font color="navy blue"> 11. 📋 List of Programs Available in More Than 5 Countries
### <font color="191970">  Global popularity of specific programs.

In [42]:
pysqldf = lambda q: sqldf(q, globals())
program = """
SELECT
  p.Program,
  COUNT(DISTINCT u.Country) AS Countries_Offered
FROM
  programs p
JOIN costs co ON p.Program_ID = co.Program_ID
JOIN universities u ON co.University_ID = u.University_ID
GROUP BY
  p.Program
HAVING
  COUNT(DISTINCT u.Country) > 5
ORDER BY
  Countries_Offered DESC;
  """
pysqldf(program)

Program  Countries_Offered
0          Computer Science                 56
1      Software Engineering                 45
2              Data Science                 45
3      Computer Engineering                 45
4       Information Systems                 36
5   Artificial Intelligence                 35
6            Data Analytics                 33
7    Information Technology                 24
8      Software Development                 21
9          Data Engineering                 16
10                 Robotics                 12
11                  Physics                  8
12              Mathematics                  7
13              Engineering                  6

🔍 Findings:

1. Computer Science is the most widespread program, offered in 56 countries, indicating its universal demand and      relevance.

2. Tech-focused programs dominate the list, reflecting global trends in digitalization, automation, and data-         driven industries.

3. The results underscore that Computer Science and its related disciplines are not only popular but also widely      accessible globally—making them strategic choices for students aiming for global academic and career mobility.

## <font color="navy blue"> 12. 💱Currency Exchange Impact
### <font color="191970"> Adjust tuition based on exchange rate for foreign students paying in local currency.

In [54]:
pysqldf = lambda q: sqldf(q, globals())
local_currency = """
SELECT
  Country,
  University,
  Tuition_USD,
  Exchange_Rate,
  ROUND(Tuition_USD * Exchange_Rate, 2) AS Local_Currency
FROM
  costs co
  join universities u ON u.University_ID = co.University_ID;
  """
pysqldf(local_currency)

Country                      University  Tuition_USD  Exchange_Rate  \
0             USA              Harvard University        55400           1.00   
1              UK         Imperial College London        41200           0.79   
2          Canada           University of Toronto        38500           1.35   
3       Australia         University of Melbourne        42000           1.52   
4         Germany  Technical University of Munich          500           0.92   
..            ...                             ...          ...            ...   
902        France        University of Strasbourg         4000           0.92   
903      Malaysia                            USIM         6800           4.65   
904  Saudi Arabia          King Faisal University         4200           3.75   
905           USA        University of Washington        50000           1.00   
906            UK        University of Nottingham        34000           0.79   

     Local_Currency  
0           55400.0  
1           32548.0  
2           51975.0  
3           63840.0  
4             460.0  
..              ...  
902          3680.0  
903         31620.0  
904         15750.0  
905         50000.0  
906         26860.0  

[907 rows x 5 columns]

🔍 Findings:

1. Countries with stronger currencies (e.g., UK, Eurozone) have tuition fees that appear lower when converted to      local currency.

2. In contrast, countries with weaker currencies (e.g., Malaysia, Saudi Arabia) see a much higher local currency      value for the same USD fee

3. This analysis is critical for international students evaluating the real cost of education abroad. It              highlights that while tuition in USD might seem comparable, the financial burden in local currency varies          drastically based on the exchange rate—making some countries more economically accessible than others.

 ## <font color="navy blue"> 13. 🏡 Find Countries Where Rent Is Over 100000 in Local Currency

In [67]:
pysqldf = lambda q: sqldf(q, globals())
query = """
SELECT
  u.Country,
  u.University,
  co.Rent_USD,
  (co.Rent_USD * co.Exchange_Rate) AS Rent_Local
FROM
  costs co
JOIN universities u ON co.University_ID = u.University_ID
WHERE
  (co.Rent_USD * co.Exchange_Rate) > 100000
ORDER BY
  Rent_Local DESC
  """
pysqldf(query)


Country                        University  Rent_USD  Rent_Local
0      Iran   Sharif University of Technology       400  16860000.0
1      Iran  Isfahan University of Technology       350  14752500.0
2      Iran               Ferdowsi University       320  13488000.0
3      Iran                 Shiraz University       300  12645000.0
4      Iran              University of Tabriz       280  11802000.0
..      ...                               ...       ...         ...
70    Japan                 Nagoya University       750    109350.0
71  Hungary            University of Pannonia       310    108577.5
72  Hungary             University of Miskolc       300    105075.0
73    Japan                   Kobe University       700    102060.0
74  Hungary       John von Neumann University       290    101572.5

[75 rows x 4 columns]

🔍 Findings:

1. All top entries are Iranian universities (e.g., Sharif University of Technology, Isfahan University of Technology).

2. Due to a very high exchange rate (1 USD ≈ 42,150 IRR), even modest rent in USD (~$300–400) translates into millions of Iranian Rials, with the top entry exceeding 16.8 million IRR.

3. While rent in USD may seem affordable, currency conversion can significantly alter its impact. Countries like Iran exhibit skyrocketing local rent costs, not due to actual rent inflation, but because of currency devaluation. This highlights the importance of analyzing costs in both USD and local currency when evaluating affordability.

4. This analysis shows how high exchange rates drastically inflate perceived costs in the local context, which is critical for budgeting and financial planning for international students.

## <font color="navy blue"> 14. 💹HIGHEST EXCHANGE RATE 
### <font color="191970"> What Currency Is Strongest Against the Dollar

In [28]:
pysqldf = lambda q: sqldf(q, globals())
highest_exchange_rate = """
SELECT
  Country,
  Exchange_Rate,
  Tuition_USD,
  (co.Tuition_USD * co.Exchange_Rate) as local
FROM
  costs as co
  JOIN universities u ON co.University_ID = u.University_ID
group by Exchange_Rate
ORDER BY
  Exchange_Rate asc
  limit 10;
"""
pysqldf(highest_exchange_rate)


Country  Exchange_Rate  Tuition_USD    local
0      Denmark           0.15         8800   1320.0
1       Kuwait           0.31         3500   1085.0
2      Bahrain           0.38         7000   2660.0
3           UK           0.79        41200  32548.0
4  Switzerland           0.89         1460   1299.4
5      Germany           0.92          500    460.0
6          USA           1.00        55400  55400.0
7    Singapore           1.34        35000  46900.0
8       Canada           1.35        38500  51975.0
9    Australia           1.52        42000  63840.0

🔍 Findings:
1. Denmark (0.15), Kuwait (0.31), and Bahrain (0.38) have the lowest exchange rates, meaning their currencies are stronger than the USD.

2. Because of the strong currency, even higher USD tuition fees result in relatively lower local currency amounts.

3. Countries like Denmark, Kuwait, and Bahrain have strong currencies, so even high tuition in USD converts to lower local currency amounts, which could influence affordability for local students. On the other hand, countries like Australia and Canada show the opposite trend, with tuition inflating when seen in local currency terms. 

## <font color="navy blue"> 15.📉                               LOWEST EXCHANGE RATE 
### <font color="191970">  What Currency Is Weakest Against the Dollar

In [30]:
pysqldf = lambda q: sqldf(q, globals())
lowest_exchange_rate = """
SELECT
  Country,
  Exchange_Rate,
  Tuition_USD,
  (co.Tuition_USD * co.Exchange_Rate) as local
FROM
  costs as co
  JOIN universities u ON co.University_ID = u.University_ID
group by Exchange_Rate
ORDER BY
  Exchange_Rate desc
LIMIT 10;
"""
pysqldf(lowest_exchange_rate)

Country  Exchange_Rate  Tuition_USD        local
0         Iran        42150.0         3000  126450000.0
1      Vietnam        24450.0         3200   78240000.0
2    Indonesia        15640.0         4200   65688000.0
3    Indonesia        15600.0         5000   78000000.0
4      Lebanon        15000.0        20000  300000000.0
5   Uzbekistan        12300.0         4200   51660000.0
6     Colombia         3950.0         3500   13825000.0
7  South Korea         1320.5         7200    9507600.0
8  South Korea         1300.0         7200    9360000.0
9      Nigeria          860.2         2500    2150500.0

🔍 Findings:

1. Countries like Iran (42,150), Vietnam (24,450), and Lebanon (15,000) have very high exchange rates.
This means 1 USD = 42,150 Iranian Rials or 15,000 Lebanese Pounds.

2. Though tuition fees in USD are moderate (e.g., $3,000–$5,000), the equivalent in local currency becomes millions.
Iran: $3,000 → 126,450,000 Rials
Lebanon: $20,000 → 300,000,000 LBP

3. Countries with the highest exchange rates face very high tuition fees in their local currencies, even when the USD value appears affordable. This reflects the currency devaluation and inflation challenges in many developing nations, which can make global education far more expensive for local students.

## <font color="navy blue"> 16. 🏅Rank Universities by Tuition Fee per Country

In [23]:
pysqldf = lambda q: sqldf(q, globals())
Rank = """
SELECT
  u.Country,
  u.University,
  p.Program,
  co.Tuition_USD,
  RANK() OVER (PARTITION BY u.Country ORDER BY co.Tuition_USD DESC) AS Tuition_Rank
FROM
  costs co
JOIN universities u ON co.University_ID = u.University_ID
JOIN programs p ON co.Program_ID = p.Program_ID;
"""
pysqldf(Rank)

Country                            University                 Program  \
0       Algeria             University of Constantine          Data Analytics   
1       Algeria  University of Science and Technology        Computer Science   
2       Algeria                    University of Oran    Computer Engineering   
3       Algeria              Badji Mokhtar University    Computer Engineering   
4       Algeria                 University of Tlemcen     Information Systems   
..          ...                                   ...                     ...   
902  Uzbekistan            Samarkand State University          Data Analytics   
903  Uzbekistan              Bukhara State University        Computer Science   
904  Uzbekistan                    Namangan Institute  Information Technology   
905  Uzbekistan              Andijan State University    Software Development   
906     Vietnam                                 HCMUT    Software Engineering   

     Tuition_USD  Tuition_Rank  
0           1300             1  
1           1200             2  
2           1100             3  
3           1000             4  
4            900             5  
..           ...           ...  
902         3800             2  
903         3500             3  
904         3400             4  
905         3200             5  
906         3200             1  

[907 rows x 5 columns]

Findings:
Each country has been assigned a Rank 1 university, representing the highest tuition fee institution for that region.